### PR TITLE
feat: Updates to `Provider` module in preparation for `InMemoryProvider`

### PR DESCRIPTION
**Pull request by maxveldink**
## This PR
- Fixes two bugs I encountered during local development (`bin/console` not loading and a memorization issue with `Configuration`
- Adds constants for `Reason` and `ErrorCode` enumerations
- Refactored `ResolutionDetails` up a module layer, and updated the client to return an `EvaluationDetails` struct, matching a callout from a spec:

> The resolution details structure is not exposed to the Application Author. It defines the data which Provider Authors must return when resolving the value of flags.

(Note at the bottom of [this section](https://openfeature.dev/specification/types#resolution-details))

### Follow-up Tasks
- Following this PR, I will PR the `InMemoryProvider` that I've been working on to satisfy a utility in [Appendix A](https://openfeature.dev/specification/appendix-a#in-memory-provider).

### CODE REVIEW

**Review by toddbaert — APPROVED** [review]

**Review by beeme1mr — APPROVED** [review]

**Comment by maxveldink** [comment]
@beeme1mr @toddbaert Before I merge in some of these PRs, I want to double-check if I need a maintainer review first or if, at the approver level, I can also merge in PRs.

**Comment by beeme1mr** [comment]
You should be able to merge whenever you're ready. Let me know if you're unable to and we'll update your permissions.

**Pull request merged**
